core/src/fragment/read_state.cc

```diff
@@ -911,6 +911,9 @@ void ReadState::get_next_overlapping_tile_sparse(const T* tile_coords) {
               &bounding_coords[dim_num], tile_subarray_end) <= 0) {
         ++search_tile_pos_;
       } else {
+        delete[] tile_subarray;
```

Comment:
Although valid, I don't think that was necessary, as control would go to the delete statements at the end of the function anyway.